.github/workflows/ci.yml

```diff
@@ -48,6 +48,15 @@ jobs:
         product: consul
         version: ${{ matrix.consul-version }}
 
+    - name: Clean Generate
+      run: |
+        echo "Checking that code generation is up-to-date"
+        go install github.com/golang/mock/mockgen@v1.6.0
+        go generate ./...
```

Comment:
Should this be captured in a make task instead of installing the specifc mockgen version only in CI?

Reply:
we could extract it into a `make` task like the others if we so desire, not opposed to the idea -- in fact I'd almost personally be inclined to just make a single `make generate` target that does all of these tasks rather than have to run each of the targets individually, personally think that it makes for better dev ergonomics